lms/static/js/dashboard/credit.js

```diff
@@ -13,5 +13,42 @@
                 }
             );
         });
+        // This event invokes credit request endpoint. It will initiate
+        // a credit request for the credit course for the provided user.
+        $(".pending-credit-btn").on("click", function(event){
+            var courseKey = $(event.target).data("course-key"),
+                username = $(event.target).data("user"),
+                provider_id = $(event.target).data("provider"),
+                postData = {
+                'course_key': courseKey,
+                'username': username
+            };
+            $.ajax({
+                url: '/api/credit/v1/provider/' + provider_id + '/request/',
+                type: 'POST',
+                headers: {
+                    'X-CSRFToken': $.cookie('csrftoken')
+                },
+                data: JSON.stringify(postData) ,
+                context: this,
+                success: function(requestData){
+                    var form = $('#credit-pending-form');
+
+                    $('input', form).remove();
+
+                    form.attr( 'action', requestData.url );
+                    form.attr( 'method', 'POST' );
+
+                    _.each( requestData.parameters, function( value, key ) {
+                        $('<input>').attr({
+                            type: 'hidden',
+                            name: key,
+                            value: value
+                        }).appendTo(form);
+                    });
+                    form.submit();
+                }
```

Comment:
If the API call fails (returns a non-200 status code), we should display an error message to the user.

@AlasdairSwan do you have any suggestions for styling an error message on the dashboard?  (It would occur after a user presses a primary action button next to a course)